wordpress-mobile/WordPress-Android
# Follow bottom sheet preparation refactor for comment snippet.

Part of #15660

This PR does some refactoring in preparation of introducing the comments snippet below the post content. The main thing done is extracting the logic for the follow conversation Bottom Sheet in a dedicated VM starting from the code that was in the ReaderCommentListViewModel before.

## To test
This PR should not introduce behaviour changes. So we need to side test with a previous version of the app that the Follow feature behaves like before.

- Go to threaded comments in the reader and tap on Follow menu
- Check you get a snackbar that allows to Enable push notifications 
- Tap on the snackbar Enable button and check you get a snackbar with an Undo button
- Tap on the Undo button and check you get an Enable snackbar again and so on
- Check that the Enable and Undo button actually activate/deactivate push notifications
- Tap on the bell menu icon and check you get the bottom sheet
- Try to enable/disable push notifications from the bottom sheet and check you get snackbar notifications with the result of the operation

## Regression Notes
1. Potential unintended areas of impact
Follow conversation feature

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit tests that were already in place

3. What automated tests I added (or what prevented me from doing so)
This is a refactor PR so no tests were added to those already present.

PR submission checklist:
